src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -557,7 +557,11 @@ def delete_ag_url_path_map_rule(resource_group_name, application_gateway_name, u
         resource_group_name, application_gateway_name, ag)
 
 def set_ag_waf_config(resource_group_name, application_gateway_name, enabled,
-                      firewall_mode=ApplicationGatewayFirewallMode.detection.value, no_wait=False):
+                      firewall_mode, no_wait=False):
+    if supported_api_version(ResourceType.MGMT_NETWORK, min_api='2016-09-01'):
+        if firewall_mode is None:
+            firewall_mode = ApplicationGatewayFirewallMode.detection.value
```

Comment:
@derekbekoe  firewall_mode  is not supported in the older api version.
Azure stack does not support application gateways although the old apiversion 2015-06-15 supports it, we left it as such in accordance with the api-version